lib/minitest/retry.rb

```diff
@@ -32,11 +41,11 @@ def self.failure_to_retry?(failures = [])
 
     module ClassMethods
       def run_one_method(klass, method_name)
-        retry_count = Minitest::Retry.retry_count
         result = super(klass, method_name)
         return result unless Minitest::Retry.failure_to_retry?(result.failures)
+        Minitest::Retry.failure_callback.call if Minitest::Retry.failure_callback
```

Comment:
I think that this process should not be executed in the case of skip. 
Therefore, I think that it is better to do after `!result.skipped?`.